src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1659,6 +1659,30 @@ private void EmitCallExpression(BoundCall call, UseKind useKind)
             FreeOptTemp(tempOpt);
         }
 
+        private bool IsReadOnlyCall(MethodSymbol method, NamedTypeSymbol methodContainingType)
+        {
+            Debug.Assert(methodContainingType.IsVerifierValue(), "only struct calls can be readonly");
+
+            if (methodContainingType.IsReadOnly && method.MethodKind != MethodKind.Constructor)
+            {
+                return true;
+            }
+
+            if (methodContainingType.IsNullableType())
+            {
+                var originalMethod = method.OriginalDefinition;
+
+                if ((object)originalMethod == this._module.Compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T_GetValueOrDefault) ||
```

Comment:
>(object)originalMethod [](start = 20, length = 22)

nit: save to a local?

Reply:
if I factor out `(object)` I get a warning on "unintended reference compare". Will keep casts. They are noops.

---
In reply to: [154443833](https://github.com/dotnet/roslyn/pull/23438#discussion_r154443833) [](ancestors = 154443833)